blueocean-dashboard/src/main/js/creation/github/GithubFlowManager.js

```diff
@@ -129,65 +126,67 @@ export default class GithubFlowManager extends FlowManager {
     }
 
     onInitialized() {
-        this.findExistingCredential();
+        this._renderCredentialsStep();
         this.setPlaceholders('Complete');
     }
 
     destroy() {
         this._cleanupListeners();
     }
 
+    getScmId() {
+        return 'github';
+    }
+
     getApiUrl() {
         // backend will default to api.github.com
         return null;
     }
 
-    findExistingCredential() {
-        return this.accessTokenManager.findExistingCredential(this.getApiUrl())
-            .then(waitAtLeast(MIN_DELAY))
```

Comment:
Yeah, it's a way to delay a Promise resolution until at least the specified time. It makes the UI feel a little smoother. It seemed the easiest way to handle this (upstream in the async layer) rather than trying to do tricks in the React layer. Any thoughts?

I formalized this as `PromiseDelayUtils` and wrote some tests for it in this refactor.